core/trino-main/src/main/java/io/trino/server/security/Authentication.java

```diff
@@ -0,0 +1,50 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.server.security;
+
+import io.trino.client.ProtocolDetectionException;
+
+import javax.ws.rs.core.MultivaluedMap;
+
+import java.util.Optional;
+
+import static io.trino.client.ProtocolHeaders.detectProtocol;
+
+public final class Authentication
+{
+    private Authentication() {}
+
+    /**
+     * When the user in the basic authentication header matches the x-trino-user header, we assume that the client does
```

Comment:
It is no longer about basic authentication

Reply:
Next commit changes method signature and it's comment to apply to new use cases.

Reply:
Actually I think I will squash those 2 commits ? WDYT ?

Reply:
Yes this was a little artificial, as I extract method to make it generic in util. Squashed.